SimTracker/TrackerMaterialAnalysis/plugins/dd4hep/DD4hep_MaterialAccountingGroup.cc

```diff
@@ -0,0 +1,236 @@
+#include <sstream>
+#include <iomanip>
+#include <string>
+#include <stdexcept>
+
+#include <TFile.h>
+#include <TH1F.h>
+#include <TProfile.h>
+#include <TCanvas.h>
+#include <TFrame.h>
+
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "DataFormats/Math/interface/Vector3D.h"
+#include "DetectorDescription/DDCMS/interface/Filter.h"
+#include "DetectorDescription/DDCMS/interface/DDFilteredView.h"
+#include "SimDataFormats/ValidationFormats/interface/MaterialAccountingStep.h"
+#include "SimDataFormats/ValidationFormats/interface/MaterialAccountingDetector.h"
+#include "DD4hep_MaterialAccountingGroup.h"
+
+DD4hep_MaterialAccountingGroup::DD4hep_MaterialAccountingGroup(const std::string& name, const cms::DDCompactView& geometry)
+    : m_name(name),
+      m_elements(),
+      m_boundingbox(),
+      m_accounting(),
+      m_errors(),
+      m_tracks(0),
+      m_counted(false),
+      m_file(nullptr) {
+  cms::DDFilter filter("TrackingMaterialGroup", {name.data(), name.size()});
+  cms::DDFilteredView fv(geometry, filter);
+  bool firstChild = fv.firstChild();
+
+  edm::LogVerbatim("TrackingMaterialAnalysis") << "Elements within: " << name;
+
+  for (const auto j : fv.specpars()) {
+    for (const auto k : j->paths) {
+      if (firstChild) {
+        std::vector<std::vector<cms::Node*>> children = fv.children(k);
+        for (auto const& path : children) {
+          cms::Translation trans = fv.translation(path);
+          GlobalPoint gp = GlobalPoint(trans.x(), trans.y(), trans.z());
+          m_elements.push_back(gp);
+          std::cout << "Adding element at (r,z) " << gp.perp() << "," << gp.z() << std::endl;
```

Comment:
Shouldn't all these `cout` statements be changed to `LogVerbatim`? Is this program only for testing?

Reply:
@cvuosalo This specific part has not changed from the existing code. Due to time constraints I'm focusing on having it working and we can address these details in future iterations